torch/nn/utils/clip_grad.py

```diff
@@ -66,6 +66,9 @@ def clip_grad_norm_(
             '`parameters` is non-finite, so it cannot be clipped. To disable '
             'this error and scale the gradients by the non-finite norm anyway, '
             'set `error_if_nonfinite=False`')
+    from torch.distributed._tensor import DTensor
+    if isinstance(total_norm, DTensor):
+        total_norm = total_norm.full_tensor()
     clip_coef = max_norm / (total_norm + 1e-6)
```

Comment:
Without an explicit call to `full_tensor()`, the `max_norm / (total_norm + 1e-6)` will use the partial `DTensor` silently, which is incorrect.